lib/Service/YAPCommon.hpp

```diff
@@ -0,0 +1,155 @@
+
+#include <stdint.h>
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/** Channel Field Length in Bytes */
+#define CHANNEL_LEN (1U)
+
+/** Checksum Field Length in Bytes */
+#define CHECKSUM_LEN (1U)
+
+/** Length of Complete Header Field */
+#define HEADER_LEN (CHANNEL_LEN + CHECKSUM_LEN)
+
+/** Data Field Length in Bytes */
+#define MAX_DATA_LEN (32U)
+
+/** Total Frame Length in Bytes */
+#define MAX_FRAME_LEN (HEADER_LEN + MAX_DATA_LEN)
+
+/** Number of Control Channel. */
+#define CONTROL_CHANNEL_NUMBER (0U)
+
+/** DLC of Heartbeat Command. */
+#define CONTROL_CHANNEL_PAYLOAD_LENGTH (11U)
+
+/** Period of Heartbeat when Synced. */
+#define HEATBEAT_PERIOD_SYNCED (5000U)
+
+/** Period of Heartbeat when Unsynced */
+#define HEATBEAT_PERIOD_UNSYNCED (1000U)
+
+/** Max length of channel name */
+#define CHANNEL_NAME_MAX_LEN (CONTROL_CHANNEL_PAYLOAD_LENGTH - 1U)
+
+/** Max number of attempts at receiving a Frame before resetting RX Buffer */
+#define MAX_RX_ATTEMPTS (MAX_FRAME_LEN)
+
+/******************************************************************************
+ * Types and Classes
+ *****************************************************************************/
+
+/**
+ * Channel Notification Prototype Callback.
+ * Provides the received data in the respective channel to the application.
+ */
+typedef void (*ChannelCallback)(const uint8_t* payload, const uint8_t payloadSize);
+
+/**
+ * Channel Definition.
+ */
+struct Channel
+{
+    char            m_name[CHANNEL_NAME_MAX_LEN]; /**< Name of the channel. */
+    uint8_t         m_dlc;                        /**< Payload length of channel */
+    ChannelCallback m_callback;                   /**< Callback to provide received data to the application. */
+
+    /**
+     * Channel Constructor.
+     */
+    Channel() : m_name(""), m_dlc(0U), m_callback(nullptr)
+    {
+    }
+};
+
+/** Data container of the Frame Fields */
+typedef union _Frame
+{
+    struct _Fields
+    {
+        /** Header */
+        union _Header
+        {
+            struct _HeaderFields
+            {
+                /** Channel ID */
+                uint8_t m_channel;
+
+                /** Frame Checksum */
+                uint8_t m_checksum;
+
+            } __attribute__((packed)) headerFields;
+
+            /** Raw Header Data*/
+            uint8_t rawHeader[HEADER_LEN];
+
+        } __attribute__((packed)) header;
+
+        /** Payload */
+        struct _Payload
+        {
+            /** Data of the Frame */
+            uint8_t m_data[MAX_DATA_LEN];
+
+        } __attribute__((packed)) payload;
+
+    } __attribute__((packed)) fields;
+
+    /** Raw Frame Data */
+    uint8_t raw[MAX_FRAME_LEN] = {0U};
+
+} __attribute__((packed)) Frame;
+
+/**
+ * Enumeration of Commands of Control Channel.
+ */
+enum COMMANDS : uint8_t

```

Comment:
It is valid since C++11 according to cppreference: https://en.cppreference.com/w/cpp/language/enum
Did I misinterpreted it, or should it not be used?

Reply:
Its ok, somehow I missed that possibility since C++11.